### ISSUE TITLE
Using as library

### ISSUE THREAD

**Issue by adamski** [ethereum/cpp-ethereum#4745]
I'm wondering if it would make sense to use some of the sub-libraries here in a cross-platform C++ project. I just want to be able to interact with an Ethereum contract, and make some transactions. 

I've started to look at making a lightweight wrapper around the [json-rpc-cpp](https://github.com/cinemast/libjson-rpc-cpp) library, with the addition of an SHA3 hash function to get the method id's. Or even simpler, just use `libcurl` to call the methods and decode the results. Basically I want to have as minimal dependencies as possible, but also a nice programming interface.

**Comment by chfast** [comment]
For that it would be better to connect to any node by JSON-RPC. So you only need a JSON-RPC client.

**Comment by adamski** [comment]
Yes I am going down that route. Only thing is I need is to be able to sign transactions with a local wallet file or private key, so that's the stuff I'm working out now, starting with RLP encoding.

**Comment by bernerdad** [comment]
adamski, have you had any luck using the cpp-ethereum libs to sign an offline transaction?  I took the second example transaction given at https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#signtransaction and tried to replicate the result (obtain the same messageHash and rawTransaction) using the cpp-ethereum libs compiled using VS2015 on Windows 10.  I cannot for the life of me get cpp-ethereum to produce the same hash and transaction as Web3 produces.
[EthTransTest.txt](https://github.com/ethereum/cpp-ethereum/files/1905142/EthTransTest.txt)

**Comment by riordant** [comment]
@adamski any progress on this? This would be very useful for a project I'm working on also

**Comment by adamski** [comment]
No. I'm working on a small C++ library to provide this functionality. I'm super busy with other projects at the moment but its in the pipeline!

**Comment by riordant** [comment]
Cool, well if you're willing to share I'm willing to contribute :)

**Comment by adamski** [comment]
@riordant Please contact me directly, its in a private repo at the moment but I'm happy to get someone else involved. Its at quite an early stage at the moment, hence not being public. I was also surprised it didn't exist yet to be honest!

**Comment by riordant** [comment]
I've contacted you on gitter.im.

**Comment by Sufi-Al-Hussaini** [comment]
@adamski , did you find time to work on this library? 

I needed similar functionality for a recent project, so I put together this demo app ([https://github.com/Sufi-Al-Hussaini/eth-rpc-demo-cpp](https://github.com/Sufi-Al-Hussaini/eth-rpc-demo-cpp)). It just demonstrates how to make rpc calls to ethereum nodes using json-rpc-cpp, so there are no user-friendly abstractions.
